components/item-table/item-table.tsx

```diff
@@ -0,0 +1,166 @@
+'use client';
+import React, { useState } from 'react';
+import { ALL_ITEMS } from '@/lib/item-store';
+import { Ingredient } from '@/lib/types/item-types';
+import ItemTableHeaderRow from './item-table-header-row';
+import { ColumnKey, tableColumns } from '@/lib/columns';
+import { Item } from '@/lib/item';
+import { SORT_DIRECTION, SortDirection } from '@/lib/types/sort-types';
+
+const ItemNameCell = (props: { item: Item }) => {
```

Comment:
In a future PR, these should be mapped to each column so we can more easily iterate over the rows without the repeated column visible check that we are currently writing.